pandas/io/common.py

```diff
@@ -153,6 +154,16 @@ def urlopen(*args, **kwargs):
     return urllib.request.urlopen(*args, **kwargs)
 
 
+def is_json(url: FilePathOrBuffer) -> bool:
+    """
+    Returns true if the given string looks like
+    json
+    """
+    json_pattern = re.compile(r"^\s*[\[{]")
```

Comment:
probably need something like the following for typing
```py
if not isinstance(url, str):
    return False
```

The pre-commit is complaining about too many new lines.